### DISCUSSION TITLE
Setter doesn't work on some properties of slider

### DISCUSSION THREAD

**Discussion by msnoshain** [AvaloniaUI/Avalonia#8746] [Q&A] [answered]
when setting background with setter:

`<Style Selector="Slider">`
`<Setter Property="Background" Value="red"/>`
`</Style>`

it doesn't work.

when setting background with inline property:

`<Slider Background="red"/>`

it works.
is this a bug or this is just how it works

**Comment by maxkatz6** [comment, marked as answer]
That's how it works
https://docs.avaloniaui.net/docs/styling/troubleshooting

**Reply by msnoshain** [reply]
What complex rules! Anyway this troubleshooting solved my problem

**Reply by timunie** [reply]
Tip: look up the original styles. Often this helps a lot.

**Reply by maxkatz6** [reply]
Most of these rules will be simplified in the 11.0. 
But this one specifically won't go anywhere, as locally set value would always override any state background otherwise.

**Reply by msnoshain** [reply]
> Tip: look up the original styles. Often this helps a lot.

can't agree more. I've seen huge improvements in the last few versions. The Selector is more and more flexible.
